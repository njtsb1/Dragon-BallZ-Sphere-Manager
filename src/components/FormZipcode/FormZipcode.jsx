import React, { useState } from 'react';
import { Box, Flex } from 'rebass'
import {
  Label,
  Input,
  Select,
  Textarea,
  Radio,
  Checkbox,
} from '@rebass/forms';
import axios from 'axios'

const FormZipcode = () => {
  const initialAddress = {
    neighborhood: '',
    publicplace: '',
    neighborhood: ''
  }
  const [ zipcode, setZipcode ] = useState('')
  const [ address, setAddress ] = useState(initialAddress)

  const fetchZipcode = async (zipcode) => {
    const zipcodeResult = await fetch(`https://viacep.com.br/ws/${zipcode}/json/`)
    const { publicplace, neighborhood, locality } = await zipcodeResult.json()
    setAddress({ publicplace, neighborhood, locality })
  }

  const handleZipcodeField = value => {
    if(value.length === 8) fetchZipcode(value)
    setZipcode(value)
  }

  return (
    <Box as='form' onSubmit={(e) => e.preventDefault()} py={3} width='500px'>
      <Flex mx={-2} mb={3}>
        <Box width={1 / 2} px={2}>
        <Label htmlFor='zipcode'>Zipcode</Label>
          <Input
            id='zipcode'
            name='zipcode'
            placeholder="Zipcode"
            onChange={({ target : { value } }) => handleZipcodeField(value)}
          />
        </Box>
        <Box width={1 / 2} px={2}>
        <Label htmlFor='neighborhood'>Neighborhood</Label>
          <Input
            id='neighborhood'
            name='neighborhood'
            placeholder="neighborhood"
            value={address.neighborhood}
            onChange={({ target : { value } }) => setAddress({...address, neighborhood: value})}
          />
        </Box>
        <Box width={1 / 2} px={2}>
        <Label htmlFor='publicplace'>Publicplace</Label>
          <Input
            id='publicplace'
            name='publicplace'
            placeholder="publicplace"
            value={address.publicplace}
            onChange={({ target : { value } }) => setAddress({...address, publicplace: value})}
          />
        </Box>
        <Box width={1 / 2} px={2}>
        <Label htmlFor='locality'>Locality</Label>
          <Input
            id='locality'
            name='locality'
            placeholder="locality"
            value={address.locality}
            onChange={({ target : { value } }) => setAddress({...address, locality: value})}
          />
        </Box>
      </Flex>
    </Box>
  );
};

export default FormZipcode;
