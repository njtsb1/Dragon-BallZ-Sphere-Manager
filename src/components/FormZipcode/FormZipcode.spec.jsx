import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import FormZipcode from './FormZipcode'

import { 
    zipcode, 
    neighborhood, 
    publicplace, 
    locality 
} from '../../mocks/address.json'


it('shold return zipcode', () => {
    addressRequest()
    cy.get('[data-testid="zipcode"]').type('11850000')
    cy.wait('@resAddress')
    cy.get('[data-testid="neighborhood"]').should('have.value', 'Jardim Francisca')
    cy.get('[data-testid="publicplace"]').should('have.value', 'Rua JoÃ£o Nagliatti')
})

const addressRequest = () => {
    cy.server()
    cy.route({
        method: 'GET',
        url: `https://viacep.com.br/ws/11850000/json`,
        response: 'fixture:address.json'
    }).as('resAddress')
}