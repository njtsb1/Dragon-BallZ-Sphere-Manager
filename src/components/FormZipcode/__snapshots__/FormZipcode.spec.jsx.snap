// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render zipcode form 1`] = `
<div>
  <form
    class="css-1ldrcmr"
  >
    <div
      class="css-1viozcy"
    >
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="zipcode"
        >
          Zipcode
        </label>
        <input
          class="css-4y926w"
          id="zipcode"
          name="zipcode"
          placeholder="Zipcode"
          type="text"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="neighborhood"
        >
          neighborhood
        </label>
        <input
          class="css-4y926w"
          id="neighborhood"
          name="neighborhood"
          placeholder="neighborhood"
          type="text"
          value="Jardim Francisca"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="publicplace"
        >
          Publicplace
        </label>
        <input
          class="css-4y926w"
          id="publicplace"
          name="publicplace"
          placeholder="publicplace"
          type="text"
          value="Street Joao Nagliatti"
        />
      </div>
    </div>
  </form>
</div>
`;
